content/en/docs/reference/access-authn-authz/service-accounts-admin.md

```diff
@@ -159,6 +160,66 @@ does not actually persist the TokenReview object into etcd.
 Hence `kubectl get tokenreview` is not a valid command.
 {{< /note >}}
 
+#### Schema for service account private claims
+
+The schema for the Kubernetes-specific claims within JWT tokens is not currently documented,
+however the relevant code area can be found in [the serviceaccount package]() in the Kubernetes codebase.
+
+[the serviceaccount package]: https://github.com/kubernetes/kubernetes/blob/d8919343526597e0788a1efe133c70d9a0c07f69/pkg/serviceaccount/claims.go#L56-L68
+
+You can inspect a JWT using standard JWT decoding tools, an example of a JWT for the
+`my-serviceaccount` ServiceAccount, bound to a Pod object named `my-pod` which is scheduled
+to the Node `my-node`, in the `default` namespace:
+
+```json
+{
+  "aud": [
+    "https://kubernetes.default.svc.cluster.local"
+  ],
+  "exp": 1729605240,
+  "iat": 1729601640,
+  "iss": "https://kubernetes.default.svc.cluster.local",
+  "jti": "aed34954-b33a-4142-b1ec-389d6bbb4936",
+  "kubernetes.io": {
+    "namespace": "default",
+    "node": {
+      "name": "my-node",
+      "uid": "646e7c5e-32d6-4d42-9dbd-e504e6cbe6b1"
+    },
+    "pod": {
+      "name": "my-pod",
+      "uid": "5e0bd49b-f040-43b0-99b7-22765a53f7f3"
+    },
+    "serviceaccount": {
+      "name": "my-serviceaccount",
+      "uid": "14ee3fa4-a7e2-420f-9f9a-dbc4507c3798"
+    }
+  },
+  "nbf": 1729601640,
+  "sub": "system:serviceaccount:default:my-serviceaccount"
+}
+```
+
+{{< note >}}
+The presence of both the `pod` and `node` claim implies that this token is bound
+to a *Pod* object. When verifying Pod bound ServiceAccount tokens, the apiserver **does not**
+verify the existence of the referenced Node object.
+{{< /note >}}
+
+Services that run outside of Kubernetes and want to perform offline validation of JWTs may
+use this schema, along with a compliant JWT validator configured with OpenID Discovery information
+from the kube-apiserver, to verify presented JWTs without requiring use of the TokenReview API.
+
+Services that verify JWTs in this way **do not verify** the claims embedded in the JWT token to be
+current and still valid.
+This means if the token is bound to an object, and that object no longer exists, the token will still
+be considered valid (until the configured token expiry).
+
+Clients that require assurance that a token's bound claims are still valid **MUST** use the TokenReview
+API to present the token to the `kube-apiserver` for it to verify and expand the embedded claims, using
+similar steps to the [Verifying and inspecting private claims](#verifying-and-inspecting-private-claims)
+section above, but with a [supported client library](/docs/reference/using-api/client-libraries/).
```

Comment:
I'm planning to add an example on using the TokenReview API to verify a claim to the `k8s.io/client-go/examples` package, which we can then link to from here (as well as via the 'Examples' link on this client libraries page).

Reply:
... unless you want to avoid a bias towards client-go and leave people to 'discover' the example by clicking through from this link :)

Reply:
opened https://github.com/kubernetes/kubernetes/pull/128268

Reply:
Do we want an example of manually inspecting/validating JWTs without TokenReview? It seems sensible for our best-practice documentation to detail how to use the proper tools, but we do also publish OpenID Discovery documents so that users *can* do this. If so, which JWT/OpenID discovery libraries are preferable? A manual HTTP request to fetch the signing information, followed by using whatever JWT handling library the apiserver already uses?

Reply:
we can link to jwt RFCs if we want

Reply:
this bit lgtm